samples/java-spring-cinema-booking-choreography/src/main/java/com/example/cinema/model/Show.java

```diff
@@ -0,0 +1,220 @@
+        Seat seat = getSeatOrThrow(seatReserved.seatNumber());
+        return new Show(id, title, seats.put(seat.number(), seat.reserved()),
+                pendingReservations.put(seatReserved.reservationId(), seatReserved.seatNumber()),
+                finishedReservations,seatReserved.availableSeatsCount());
+    }
+
+    public Show applyReservationPaid(ShowEvent.SeatReservationPaid seatReservationPaid) {
+        Seat seat = getSeatOrThrow(seatReservationPaid.seatNumber());
+        String reservationId = seatReservationPaid.reservationId();
+        FinishedReservation finishedReservation = new FinishedReservation(reservationId, seat.number()/*, CONFIRMED*/);
+        return new Show(id, title, seats.put(seat.number(), seat.paid()),
+                pendingReservations.remove(reservationId),
+                finishedReservations.put(reservationId, finishedReservation),availableSeats());
+
+    }
+
+    public Show applyReservationCancelled(ShowEvent.SeatReservationCancelled seatReservationCancelled) {
+        Seat seat = getSeatOrThrow(seatReservationCancelled.seatNumber());
+        String reservationId = seatReservationCancelled.reservationId();
+        FinishedReservation finishedReservation = new FinishedReservation(reservationId, seat.number()/*, CANCELLED*/);
+        return new Show(id, title, seats.put(seat.number(), seat.available()),
+                pendingReservations.remove(reservationId),
+                finishedReservations.put(reservationId, finishedReservation),seatReservationCancelled.availableSeatsCount());
+    }
+
+
+    private Seat getSeatOrThrow(int seatNumber) {
+        return seats.get(seatNumber).getOrElseThrow(() -> new IllegalStateException("Seat not found %s".formatted(seatNumber)));
+    }
+
+    public Option<Seat> getSeat(int seatNumber) {
+        return seats.get(seatNumber);
+    }
+
+    public enum SeatReservationStatus {
+      STARTED, SEAT_RESERVED, WALLET_CHARGE_REJECTED, WALLET_CHARGED, COMPLETED, SEAT_RESERVATION_FAILED, WALLET_REFUNDED, SEAT_RESERVATION_REFUNDED
+    }
+
+    public enum SeatStatus {
+      AVAILABLE, RESERVED, PAID
+    }
+
+    public record FinishedReservation(String reservationId, int seatNumber) { }
+
+    public record Reservation(String reservationId, String showId, String walletId, BigDecimal price) {}
+
+    public record Seat(int number, SeatStatus status, BigDecimal price) {
+      @JsonIgnore
+      public boolean isAvailable() {
+        return status == AVAILABLE;
+      }
+
+      public Seat reserved() {
+        return new Seat(number, SeatStatus.RESERVED, price);
+      }
+
+      public Seat paid() {
+        return new Seat(number, SeatStatus.PAID, price);
+      }
+
+      public Seat available() {
+        return new Seat(number, AVAILABLE, price);
+      }
+    }
+
+    public record SeatReservation(String reservationId, String showId, int seatNumber, String walletId, BigDecimal price,
```

Comment:
There are a lot of unused methods in this record; maybe in others as well, since it's a simplified version of the original app.